server/src/test/java/org/elasticsearch/indices/state/OpenCloseIndexIT.java

```diff
@@ -377,8 +378,13 @@ public void testTranslogStats()  {
 
         IndicesStatsResponse stats = client().admin().indices().prepareStats(indexName).clear().setTranslog(true).get();
         assertThat(stats.getIndex(indexName), notNullValue());
-        assertThat(stats.getIndex(indexName).getPrimaries().getTranslog().estimatedNumberOfOperations(), equalTo(
-            softDeletesEnabled ? uncommittedOps : nbDocs));
+        // If soft-deletes is enabled, we will trim translog above the local checkpoint of the safe commit immediately.
+        // However, if the translog durability is async, the last commit might not be the safe commit as the local checkpoint
+        // won't advance until translog is synced. We can't check for translog stats in this case.
+        if (translogDurability != Translog.Durability.ASYNC || softDeletesEnabled == false) {
```

Comment:
Can't we busy assert here in the async durability case?  The max sync interval setting we use in tests seems to be `5s` so that should work?

Reply:
Good suggestion. I pushed https://github.com/elastic/elasticsearch/pull/45809/commits/2a63de94e4eeb662c71a3b23c1276d8828bff5f4.